### DISCUSSION TITLE
CSC Shipping Address Validation

### DISCUSSION THREAD

**Discussion by alexsuciuinfo** [SalesforceCommerceCloud/sfcc-ci#418] [Team Posts]
@SalesforceCommerceCloud/community 

I am trying to validate an address in Customer Service Center using the Shop API Hook **dw.ocapi.shop.basket.shipment.shipping_address.afterPUT**, but the corresponding script is not getting executed. Has anyone been able to use this hook successfully?

**Comment by sholsinger** [comment]
First, I’d say You should do all validation in a before* hook as it is too
late to reject the value in an after* hooks.

Secondly, i’d suggest you check error logs for a compilation error.

**Comment by alexsuciuinfo** [comment]
I have no compilation erros.
The main idea is that none of the **beforePUT/afterPUT** methods are executed: 

- dw.ocapi.shop.basket.shipment.shipping_address.beforePUT
- dw.ocapi.shop.basket.shipment.shipping_address.afterPUT
- dw.ocapi.shop.basket.billing_address.beforePUT
- dw.ocapi.shop.basket.billing_address.afterPUT

Instead, the **modifyPUTResponse** are working:

- dw.ocapi.shop.basket.billing_address.modifyPUTResponse
- dw.ocapi.shop.basket.shipment.shipping_address.modifyPUTResponse

And I was wondering if someone was able to use them successfully.

**Comment by kkuhikar** [comment]
> I have no compilation erros.
> The main idea is that none of the beforePUT/afterPUT methods are executed:
> 
> dw.ocapi.shop.basket.shipment.shipping_address.beforePUT
> dw.ocapi.shop.basket.shipment.shipping_address.afterPUT
> dw.ocapi.shop.basket.billing_address.beforePUT
> dw.ocapi.shop.basket.billing_address.afterPUT
> Instead, the modifyPUTResponse are working:
> 
> dw.ocapi.shop.basket.billing_address.modifyPUTResponse
> dw.ocapi.shop.basket.shipment.shipping_address.modifyPUTResponse
> And I was wondering if someone was able to use them successfully.

@alexsuciuinfo, I am having similar outcome as you mentioned.

I tried with a different Shop API hook as well. Same as with the **beforePUT/afterPUT** methods, I am observing that the **beforePATCH/afterPATCH** methods are not getting executed:
- dw.ocapi.shop.basket.item.beforePATCH
- dw.ocapi.shop.basket.item.afterPATCH

And, the **modifyPATCHResponse** is working:
- dw.ocapi.shop.basket.item.modifyPATCHResponse

Please advise how to use the aforementioned methods successfully.

**Comment by kkuhikar** [comment]
**Update**
I tried today with Shop API version 19.5, and now I am able to use the before/after methods.

**Comment by sholsinger** [comment]
Be sure to check the version of the documentation you're working with. For
example, there are _three_ different documentation versions:

- *DOC1 PIG Instances*: (Currently at v19.6) https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.5/usage/OpenCommerceAPI.html?cp=0_12
- *DOC2: SIG Instances* (Currently at v19.6) https://documentation.b2c.commercecloud.salesforce.com/DOC2/topic/com.demandware.dochelp/OCAPI/19.5/usage/OpenCommerceAPI.html?cp=0_12
- *DOC3: Early Access Instances* (Currently at v19.7) https://documentation.b2c.commercecloud.salesforce.com/DOC3/topic/com.demandware.dochelp/OCAPI/19.5/usage/OpenCommerceAPI.html?cp=0_12
